### ISSUE TITLE
AMI Builder (EBS): encrypt_boot encrypts ami again if the source ami is already encrypted

### ISSUE THREAD

**Issue by ervoiox** [hashicorp/packer#4923]
Hi,

I'm not sure if this is intended behavior - but if you have `encrypt_boot` enabled and the source AMI is already encrypted - packer will go through the process to encrypt the AMI again... when it seems like that isn't necessary seeing as the source AMI is already encrypted.

On the other hand, setting `encrypt_boot` to false with a encrypted source ami, does nothing, but the boot is still encrypted. So a work around if you already know your source ami is encrypted is to leave `encrypted_boot` to false.

I think `encrypt_boot` should not do anything if the source ami is already encrypted and the KMS key used has not changed. Then the documentation then could be updated so that `encrypted_boot` only works on non encrypted source AMIs or the KMS key used for encryption changes.

**Comment by mwhooker** [comment]
Hi @ervoiox 
thanks for the feedback.

I think we want to encrypt any amis we create if `encrypt_boot` is enabled. Am I misunderstanding your question?

**Comment by ervoiox** [comment]
Hi @mwhooker,

I am saying that if the source ami is already encrypted and the KMS key hasn't changed - does it really need to go through the `encrypt_boot` process again?

**Comment by rickard-von-essen** [comment]
I think the docs are clear. And this works as expected.

**Comment by mwhooker** [comment]
Thanks again for the suggestion. We try not to make packer too smart, because it invariably creates edge cases which can be difficult debug. I think we're going to leave it the way it is for now.

**Comment by thomasbiddle** [comment]
I think at the least, this could be made more clear in the documentation. I was confused as well. @rickard-von-essen @mwhooker

**Comment by rickard-von-essen** [comment]
Let's add a note in the docs.

**Comment by SwampDragons** [comment]
@thomasbiddle or @ervoiox if you have time could either of you give me an example of a documentation note you think would have clarified this issue for a new user? I'm not totally sure I see how this is unexpected, so I'm struggling to anticipate what would help.  Thanks!

**Comment by rickard-von-essen** [comment]
From https://www.packer.io/docs/builders/amazon-ebs.html#encrypt_boot :

> (discarding the initial unencrypted AMI in the process). 

This is slightly inaccurate, a snapshot taken from a encrypted ebs volume will be encrypted (with the same KMS key).

**Comment by thomasbiddle** [comment]
@SwampDragons 

Current version from https://www.packer.io/docs/builders/amazon-ebs.html#encrypt_boot

> **encrypt_boot** (boolean) - Instruct packer to automatically create a copy of the AMI with an encrypted boot volume (discarding the initial unencrypted AMI in the process). Default `false.`

Maybe something like this would be more clear:

**encrypt_boot** (boolean) - Instruct Packer to automatically create a copy of the AMI with an encrypted boot volume. Packer will always run this operation, even if the base AMI has an encrypted boot volume to start with. Default `false.`

It's a bit wordy - but I'm not copywriter :-)

**Comment by SwampDragons** [comment]
Awesome. I'd rather wordy than inaccurate! Thanks for the copy!

**Comment by SwampDragons** [comment]
@thomasbiddle this seem okay to you? https://github.com/hashicorp/packer/pull/5132. I left in the parenthetical about deleting the unencrypted volume.

**Comment by thomasbiddle** [comment]
@SwampDragons Added in a review
